res/enkf/queue_config.py

```diff
@@ -135,6 +135,8 @@ def job_script(self):
 
     @property
     def driver(self):
+        if not self.queue_system:
```

Comment:
This kind of configuration validation has traditionally been the responsibility of `config_parser`. Could you please elaborate on this change?

Reply:
Added this to have a reasonable error when there is no QUEUE_SYSTEM. We first attempted adding it to the required schema items used in config_validate. But it seemed to not be able to cover the case where it is there in the site-config, as it will then be required in the user config as well. Perhaps we misunderstood something though.

Reply:
That is a good intention. However, since all the other validation of the entire ERT configuration is done together with the `config_parser` I think we should strive to also do this one within that framework.

It is on this line the queue system is extracted: https://github.com/equinor/ert/blob/37ecdbe7d6c0159f5b91ef66b9a37d38d7d198a1/libres/lib/enkf/queue_config.cpp#L268.

It is not as easy as always requiring the keyword to be there, as it might not be present in either the site-configuration or the user configuration. However, the queue configuration already have the flag `user_mode` to separate between parsing of the site-config and the user config and then we might as well use it...

I would suggest that we either:
- require the `site-config` to contain `QUEUE_SYSTEM`. I think deploying ERT without a default queue system is to put an unnecessary burden on all of the users. And if the default site-config then contains a `QUEUE_SYSTEM` as this PR introduces I do not consider this to be a problematic change.
- follow up the block extracting the queue system with an `else if` that raise a util abort if this is the user mode and the queue system is still not set.

I'm fine with either solution, but I think that is where the responsibility in the code should reside.

Reply:
We went for option 1. Discovered (after a lot of debugging) that validation was not enabled for the site-config, so we also had to enable that.